exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -216,6 +217,20 @@ public void eval() {
   }
 
   @FunctionTemplate(names = {"char_length", "character_length", "length"}, scope = FunctionScope.SIMPLE, nulls = NullHandling.NULL_IF_NULL)
+  public static class DummyCharLength implements DrillSimpleFunc {
```

Comment:
As you can see, there are a bunch of dummy functions added in. 
As here:
https://github.com/hsuanyi/incubator-drill/blob/DRILL-4372_For_Review/exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/TypeInferenceUtils.java#L101

If there is no a dummy, this line cannot find a DrillFuncHolder which matches the given SqlOperatorBinding. 

Their bodies are empty and it is fine because later in DrillOptiq, the functions which will actually be used will be chosen.